app/payment-link-v2/amount/get-index.controller.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+// Custom dependencies
+const { response } = require('../../utils/response')
+const { getSessionVariable } = require('../../utils/cookie')
+const paths = require('../../paths')
+const replaceParamsInPath = require('../../utils/replace-params-in-path')
+const { NotFoundError } = require('../../errors')
+
+module.exports = (req, res, next) => {
+  const product = req.product
+  const data = {
+    productExternalId: product.external_id,
+    productName: product.name,
+    productDescription: product.description,
+    productReferenceEnabled: product.reference_enabled,
+    captchaChallengeFailed: req.captchaChallengeFailed
+  }
+
+  if (product.price) {
+    return next(new NotFoundError('Attempted to access amount page with a product that already has a price.'))
+  }
+
+  const sessionAmount = getSessionVariable(req, 'amount')
+
+  if (sessionAmount) {
+    data.backLinkHref = replaceParamsInPath(paths.paymentLinksV2.confirm, product.external_id)
+    data.productAmount = (sessionAmount / 100).toFixed(2)
+    return response(req, res, 'amount/amount', data)
+  }
+
+  if (product.reference_enabled) {
+    const sessionReferenceNumber = getSessionVariable(req, 'referenceNumber')
+    if (!sessionReferenceNumber) {
+      data.backLinkHref = replaceParamsInPath(paths.paymentLinksV2.reference, product.external_id)
+      return response(req, res, 'amount/amount', data)
+    }
+  }
+
+  if (!product.reference_enabled) {
```

Comment:
You can get rid of this if condition, as it will always be true. You can then also remove the `return next(...)` at the end as it will never be hit.

We're also always calling `return response(req, res, 'amount/amount', data)` so that can be moved down to the bottom of the controller to avoid repetition.

Reply:
Done - thanks.

We had to add a final `else` as the different conditions change the `back` link.